astropy/config/tests/test_configs.py

```diff
@@ -37,26 +37,122 @@ def test_paths():
     assert "testpkg" in paths.get_cache_dir(rootname="testpkg")
 
 
+@pytest.fixture
+def isolated_setup(monkeypatch):
+    # ignore global state of the test session
+    monkeypatch.delenv("ASTROPY_CACHE_DIR", raising=False)
+    monkeypatch.delenv("ASTROPY_CONFIG_DIR", raising=False)
+    monkeypatch.delenv("XDG_CACHE_HOME", raising=False)
+    monkeypatch.delenv("XDG_CONFIG_HOME", raising=False)
+
+    monkeypatch.setattr(paths.set_temp_cache, "_temp_path", None)
+    monkeypatch.setattr(paths.set_temp_config, "_temp_path", None)
+
+
 @pytest.mark.parametrize(
-    "environment_variable,func",
+    "env_var_template",
     [
-        # Regression test for #17514 - XDG_CACHE_HOME had no effect
-        pytest.param("XDG_CACHE_HOME", paths.get_cache_dir_path, id="cache"),
-        pytest.param("XDG_CONFIG_HOME", paths.get_config_dir_path, id="config"),
+        pytest.param("XDG_{}_HOME", id="xdg"),
+        pytest.param("ASTROPY_{}_DIR", id="astropy"),
     ],
 )
-def test_xdg_variables(monkeypatch, tmp_path, environment_variable, func):
-    config_dir = tmp_path / "astropy"
-    config_dir.mkdir()
+@pytest.mark.parametrize(
+    "dir_type, func",
+    [
+        pytest.param("CACHE", paths.get_cache_dir_path, id="cache"),
+        pytest.param("CONFIG", paths.get_config_dir_path, id="config"),
+    ],
+)
+@pytest.mark.usefixtures("isolated_setup")
+def test_env_variables(monkeypatch, tmp_path, env_var_template, dir_type, func):
+    environment_variable = env_var_template.format(dir_type)
+    target_dir = tmp_path / "astropy"
+    target_dir.mkdir()
     monkeypatch.setenv(environment_variable, str(tmp_path))
-    assert func() == config_dir
+    assert func() == target_dir
 
 
-def test_set_temp_config(tmp_path, monkeypatch):
+@pytest.mark.parametrize(
+    "astropy_env_var, xdg_env_var, cls, func",
+    [
+        pytest.param(
+            "ASTROPY_CACHE_DIR",
+            "XDG_CACHE_HOME",
+            paths.set_temp_cache,
+            paths.get_cache_dir_path,
+            id="cache",
+        ),
+        pytest.param(
+            "ASTROPY_CONFIG_DIR",
+            "XDG_CONFIG_HOME",
+            paths.set_temp_config,
+            paths.get_config_dir_path,
+            id="config",
+        ),
+    ],
+)
+@pytest.mark.usefixtures("isolated_setup")
+def test_env_variables_priority(
+    monkeypatch, tmp_path, astropy_env_var, xdg_env_var, cls, func
+):
+    # ASTROPY_* environment variables should have priority over XDG_*
+    astropy_target_dir = tmp_path / "astropy" / "astropy"
+    astropy_target_dir.mkdir(parents=True)
+    monkeypatch.setenv(astropy_env_var, str(tmp_path / "astropy"))
+
+    xdg_target_dir = tmp_path / "xdg" / "astropy"
+    xdg_target_dir.mkdir(parents=True)
+    monkeypatch.setenv(xdg_env_var, str(tmp_path / "xdg"))
+
+    monkeypatch.setattr(cls, "_temp_path", None)
```

Comment:
Why is this needed if you are using the `isolated_setup` fixture?

Reply:
That's just accidental left overs. Thanks !